juniper_official/Predictive-Maintenance/check-optics-tx-rx-power-ml.rule

```diff
@@ -0,0 +1,189 @@
+/*
+ * Checks for anomalies in optic module tx and rx 
+ *  
+ */
+healthbot {
+    topic linecard.optical {
+        rule check-optics-tx-rx-power-ml {
+            /*
+             * Monitors anomalies in optic module tx and rx
+             *
+             * Use interface-name as key for rule.
+             */		
+            keys interface-name;
+            description "tx and rx anomaly for optic  interface";
+            /*
+             * Sensor configuration to get data from network devices.
+             */			
+            sensor optics-power {
+                iAgent {
+                    file interfaceOpticsDiagnostics-tx-rx.yml;
+                    table interfaceOpticsDiagnostics-tx-rx;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined with variables and constant values
+             */			
+            field interface-name {
+                sensor optics-power {
+                    path name;
+                }
+                type string;
+                description "interface name";
+            }
+            field rx-power {
+                sensor optics-power {
+                    path rx-signal-avg-optical-power-dbm;
+                }
+                type float;
+                description "optic rx power in db";
+            }
+            field rx-power-anomaly {
+                formula {
+                    anomaly-detection {
+                        algorithm 3sigma;
+                        learning-period 7days;
+                        pattern-periodicity 1days;
+                        field-name "$rx-power";
+                    }
+                }
+                type integer;
+                description "rx power anomaly";
+            }
+            field tx-power {
+                sensor optics-power {
+                    path laser-output-power-dbm;
+                }
+                type float;
+                description "optic tx power in db";
+            }
+            field tx-power-anomaly {
+                formula {
+                    anomaly-detection {
+                        algorithm 3sigma;
+                        learning-period 7days;
+                        pattern-periodicity 1days;
+                        field-name "$tx-power";
+                    }
+                }
+                type integer;
+                description "tx power anomaly";
+            }
+            /*
+             * Anomaly detection logic.
+             */			
+            trigger tx-rx-anomaly {
```

Comment:
Please pick a more intuitive name for the trigger. Avoid using anomaly in the trigger.
Maybe something like tx-rx-power?